javascript/ql/src/Security/CWE-798/HardcodedCredentials.ql

```diff
@@ -22,8 +22,18 @@ where
   // use source value in message if it's available
   if source.getNode().asExpr() instanceof ConstantString
   then
-    value = "The hard-coded value \"" + source.getNode().asExpr().(ConstantString).getStringValue() +
-        "\""
+    exists(string val, Sink sinkNode |
+      sinkNode = sink.getNode().(Sink) and
+      val = source.getNode().asExpr().(ConstantString).getStringValue()
+    |
+      (
+        (sinkNode.(DefaultCredentialsSink).getKind() = "password" or sinkNode.(DefaultCredentialsSink).getKind() = "key")
```

Comment:
Perhaps pull out `sinkNode.(DefaultCredentialsSink).getKind()` into a variable for brevity?